packages/web/src/pages/edit-page/EditTrackPage.tsx

```diff
@@ -86,8 +77,23 @@ export const EditTrackPage = (props: EditPageProps) => {
         }
       })
     } else {
-      dispatch(editTrack(metadata.track_id, metadata))
-      dispatch(push(metadata.permalink))
+      if (!track) return
+      updateTrack(
+        {
+          trackId: metadata.track_id,
+          userId: track.owner_id,
+          metadata,
+          coverArtFile:
+            metadata.artwork && 'file' in metadata.artwork
+              ? (metadata.artwork.file as File)
+              : undefined
+        },
+        {
+          onSuccess: () => {
```

Comment:
what happens in the meantime while this is firing off? do we want to put a loading state somewhere?

Reply:
did a bunch of fixes to make optimistic updates work, and now we just go straight to the track page while the update is in progress